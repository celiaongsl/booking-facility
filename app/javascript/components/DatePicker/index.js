import React from "react";

import { makeStyles } from "@material-ui/core/styles";
import {
  KeyboardDatePicker,
  MuiPickersUtilsProvider,
} from "@material-ui/pickers";
import DateFnsUtils from "@date-io/date-fns";

const useStyles = makeStyles({
  root: {
    margin: 10,
  },
});

const DatePicker = (props) => {
  const { value, onChange } = props;
  const classes = useStyles();
  return (
    <MuiPickersUtilsProvider utils={DateFnsUtils}>
      <KeyboardDatePicker
        className={classes.root}
        disableToolbar
        variant="inline"
        format="MM/dd/yyyy"
        margin="normal"
        label="MM/dd/yyyy"
        id="date-picker-inline"
        value={value}
        onChange={onChange}
        KeyboardButtonProps={{
          "aria-label": "change date",
        }}
      />
    </MuiPickersUtilsProvider>
  );
};

export default DatePicker;
