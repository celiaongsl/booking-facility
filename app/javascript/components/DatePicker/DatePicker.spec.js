import React from "react";
import DatePicker from "./";
import { render, fireEvent } from "@testing-library/react";

describe("<DatePicker />", () => {
  const setup = () => {
    const utils = render(<DatePicker />);
    const datepickerInput = utils.getByLabelText("change date");
    return {
      datepickerInput,
      ...utils,
    };
  };

  it("renders the component", () => {
    const container = render(<DatePicker />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("it should display the right date", () => {
    const { datepickerInput } = setup();

    console.log(datepickerInput.value)
    fireEvent.change(datepickerInput, { target: { value: "10-12-2020" } });
    expect(datepickerInput.value).toBe("10-12-2020");
    console.log(datepickerInput.value)
  });
});
