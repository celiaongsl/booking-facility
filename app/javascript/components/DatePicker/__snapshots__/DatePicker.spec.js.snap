// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePicker /> renders the component 1`] = `
<DocumentFragment>
  <div
    aria-label="date-picker"
    class="MuiFormControl-root MuiTextField-root makeStyles-root-1 MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="date-picker-inline"
      id="date-picker-inline-label"
    >
      Booking Date
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
        id="date-picker-inline"
        type="text"
        value="02/26/2021"
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
      >
        <button
          aria-label="change date"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
